tick/base/src/interruption.h

```diff
@@ -19,31 +21,40 @@
  */
 class Interruption : public std::exception {
  private:
-    static std::atomic<bool> flag_interrupt;
+    static std::atomic<bool>& get_flag_interrupt() {
+        static std::atomic<bool> flag_interrupt;
+        return flag_interrupt;
+    }
 
  public:
     //! @brief Simple constructor
     Interruption() {}
 
-    const char *what() const noexcept override;
+    DLL_PUBLIC const char *what() const noexcept override;
 
     //! \cond
     //! @brief Notifies an interruption has been detected (not to be called directly)
-    inline static void set() { flag_interrupt = true; }
+    inline static void set() {
+        auto& flag_interrupt = get_flag_interrupt();
+        flag_interrupt = true;
```

Comment:
the static member was replaced by a static function containing a static member as static members across DLLs don't work so well, it could be fine to just have 


````
get_flag_interrupt() = true;
```